src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt

```diff
@@ -1669,6 +1671,93 @@ class PlantingSiteStore(
+
+    val messages = mutableListOf<String>()
+
+    dslContext.transaction { _ ->
+      val simplePlantingSiteIds =
+          dslContext
+              .select(PLANTING_SITES.ID)
+              .distinctOn(PLANTING_SITES.ID)
+              .from(PLANTING_SITES)
+              .where(PLANTING_SITES.BOUNDARY.isNotNull)
+              .andNotExists(
+                  DSL.selectOne()
+                      .from(PLANTING_ZONES)
+                      .where(PLANTING_ZONES.PLANTING_SITE_ID.eq(PLANTING_SITES.ID)))
+              .orderBy(PLANTING_SITES.ID)
+              .fetch(PLANTING_SITES.ID.asNonNullable())
+
+      for (siteId in simplePlantingSiteIds) {
+        try {
+          val now = clock.instant()
+
+          log.info("Migrating simple planting site $siteId to detailed")
+
+          val existingSite = fetchSiteById(siteId, PlantingSiteDepth.Site)
+          val boundary = existingSite.boundary!!.toMultiPolygon()
+
+          // This calculates the grid origin and area.
+          val newSite =
+              NewPlantingSiteModel.create(
+                  boundary,
+                  existingSite.description,
+                  name = existingSite.name,
+                  organizationId = existingSite.organizationId,
+              )
+
+          if (newSite.areaHa == null || newSite.areaHa.equalsIgnoreScale(BigDecimal.ZERO)) {
+            messages.add("Planting site $siteId (${existingSite.name}) area too small to convert")
+            continue
+          }
+
+          val zone =
+              NewPlantingZoneModel.create(
+                  boundary = boundary,
+                  name = zoneName,
+                  plantingSubzones = emptyList(),
+              )
+          val subzone =
+              NewPlantingSubzoneModel.create(
+                  boundary = boundary,
+                  fullName = "$zoneName-$subzoneName",
+                  name = subzoneName,
+              )
+
+          with(PLANTING_SITES) {
+            dslContext
+                .update(PLANTING_SITES)
+                .set(AREA_HA, newSite.areaHa)
+                .set(GRID_ORIGIN, newSite.gridOrigin)
+                .set(MODIFIED_BY, currentUser().userId)
+                .set(MODIFIED_TIME, now)
+                .where(ID.eq(siteId))
+                .execute()
+          }
+
+          val siteHistoryId = insertPlantingSiteHistory(newSite, newSite.gridOrigin!!, now, siteId)
+          val zoneId = createPlantingZone(zone, siteId)
+          val zoneHistoryId = insertPlantingZoneHistory(zone, siteHistoryId, zoneId)
+          val subzoneId = createPlantingSubzone(subzone, siteId, zoneId)
+          insertPlantingSubzoneHistory(subzone, zoneHistoryId, subzoneId)
+
+          messages.add("Migrated planting site $siteId (${existingSite.name})")
+        } catch (e: Exception) {
+          log.error("Unable to migrate planting site $siteId", e)
+          messages.add("Unable to migrate planting site $siteId: ${e.message}")
+          break
```

Comment:
We're not throwing an error here, which means that the error gets added to `messages`, which then gets set to `redirectAttributes.successDetails` in the controller, and the controller would still display `Successfully migrated simple planting sites`, which seems to be confusing since the messages below it would say `Unable to migrate planting site <bla>`

Reply:
Yeah, fair point, this is shoddy.

Reply:
And, if it's a one time use and you're the most likely person to use it then it might not need fixing...